code/modules/clothing/head/perceptomatrix.dm

```diff
@@ -0,0 +1,204 @@
+
+	// some-fucking-how, RemoveElement fails to find the element in below proc if i try to remove it. so whatever.
+	AddElement(/datum/element/wearable_client_colour, /datum/client_colour/perceptomatrix, ITEM_SLOT_HEAD, forced = TRUE)
+
+/obj/item/clothing/head/helmet/perceptomatrix/proc/update_anomaly_state()
+
+	// If the core isn't installed, or it's temporarily deactivated, disable special functions.
+	if(!core_installed)
+		clothing_flags = initial(clothing_flags) & ~CASTING_CLOTHES
+		detach_clothing_traits(clothing_traits)
+		QDEL_LIST(active_components)
+		return
+
+	clothing_flags = initial(clothing_flags)
+	attach_clothing_traits(initial(clothing_traits))
+
+	LAZYADD(active_components, AddComponent(/datum/component/wearertargeting/earprotection, list(ITEM_SLOT_HEAD)))
+	LAZYADD(active_components, AddComponent(
+		/datum/component/anti_magic, \
+		antimagic_flags = MAGIC_RESISTANCE_MIND, \
+		inventory_flags = ITEM_SLOT_HEAD, \
+	))
+
+	update_icon_state()
+
+/obj/item/clothing/head/helmet/perceptomatrix/Destroy(force)
+	QDEL_NULL(transform_action)
+	QDEL_NULL(active_components)
+	return ..()
+
+/obj/item/clothing/head/helmet/perceptomatrix/examine(mob/user)
+	. = ..()
+	if (!core_installed)
+		. += span_warning("It requires a hallucination anomaly core in order to function.")
+
+/obj/item/clothing/head/helmet/perceptomatrix/item_action_slot_check(slot, mob/user, datum/action/action)
+	return slot & ITEM_SLOT_HEAD
+
+/obj/item/clothing/head/helmet/perceptomatrix/update_icon_state()
+	icon_state = base_icon_state + (core_installed ? "" : "_inactive")
+	worn_icon_state = base_icon_state + (core_installed ? "" : "_inactive")
+	return ..()
+
+/obj/item/clothing/head/helmet/perceptomatrix/attackby(obj/item/weapon, mob/user, params)
+	if (!istype(weapon, /obj/item/assembly/signaler/anomaly/hallucination))
+		return ..()
+	balloon_alert(user, "inserting...")
+	if (!do_after(user, delay = 3 SECONDS, target = src))
+		return
+	qdel(weapon)
+	core_installed = TRUE
+	update_anomaly_state()
+	update_appearance(UPDATE_ICON_STATE)
+	playsound(src, 'sound/machines/crate/crate_open.ogg', 50, FALSE)
+
+/// Pre-activated polymorph belt
+/obj/item/clothing/head/helmet/perceptomatrix/functioning
+	core_installed = TRUE
+
+/datum/action/cooldown/spell/pointed/percept_hallucination
+	name = "Hallucinate"
+	desc = "Redirect perceptual energies towards a target, staggering them. Temporarily disables the perceptomatrix helmet."
+	button_icon_state = "blind"
+	ranged_mousepointer = 'icons/effects/mouse_pointers/blind_target.dmi'
+
+	sound = 'sound/items/weapons/emitter2.ogg'
+	school = SCHOOL_PSYCHIC
+	cooldown_time = 25 SECONDS
+
+	invocation_type = INVOCATION_NONE
+	spell_requirements = NONE
+	antimagic_flags = MAGIC_RESISTANCE_MIND
+
+	active_msg = "You prepare to zap a target with hallucinations..."
+
+	/// The amount of blurriness to apply
+	var/eye_blur_duration = 7 SECONDS
+	/// The amount of stagger to apply
+	var/stagger_duration = 3 SECONDS
```

Comment:
Honestly, given the relatively huge cooldown and losing the benefits of the helmet while on cooldown, you could stand to either make this stagger longer or alternatively make the cooldown lower. No need to be _too_ conservative with it, it does use an anomaly core, so it has grounds for being a bit spammable for what it applies (though the more powerful you make it, the lower the number of hallucination cores you should let be craftable in the first place)

Reply:
The helmet disable was ripped off because I couldn't get it to work properly

Reply:
Cool beans. It's probably fine as is then.